packages/instrumentation-aws-sdk/src/utils.ts

```diff
@@ -28,9 +29,9 @@ export const normalizeV3Request = (
     commandInput: Record<string, any>,
     region: string
 ): NormalizedRequest => {
-    const commandName = toCamelCase(removeSuffixFromStringIfExists(commandNameWithSuffix, 'Command'));
+    const commandName = removeSuffixFromStringIfExists(commandNameWithSuffix, 'Command');
     return {
-        serviceName: serviceName?.toLowerCase(),
+        serviceName,
```

Comment:
In my test, whether it is `clientConfig.serviceId` or `removeSuffixFromStringIfExists(awsExecutionContext.clientName, 'Client')`, they are both already `DynamoDB`:

https://github.com/aspecto-io/opentelemetry-ext-js/blob/243e19bcebb5f95df5b4a68cbf675aa6351ca3fb/packages/instrumentation-aws-sdk/src/aws-sdk.ts#L313-L314

Reply:
That is the case for services that don't have spaces in them (like DynamoDB), but it will produce different results for services like `Route 53`:
clientName is: `Route53Client` which will produce the value `"Route53"` while the `serviceId` is `"Route 53"` with a space.
That's really annoying but not sure we have any better alternative, as the the sdk is written a way which makes it really hard to extract this info correctly.
What do you think?

Reply:
I think the spec not saying anything too detailed about this means it's up to us 🙂 I think to make it easy on ourselves, we should just stick to `api.serviceId` in V2 and `clientConfig.serviceId` like we have here in V3.

I tested this on V2 and `api.serviceId` will produce `Route 53` (with the space) so it matches with the current `clientConfig.serviceId`.

This still supports my original comment that we can delete the `removeSuffixFromStringIfExists()` call... But I'm not too familiar with the AWS JS SDKs so I think it's fine to leave it for now in case that code path gets hit 🙂 

Maybe an action item would be to add `Route 53` to the spec examples so other SDKs can follow this decision?

Reply:
aws-sdk v3 is written in a very non-instrumentation-friendly fashion.
The config is only available on the `smithy-client`, but instrumentation is not possible there since the operation (or commandName) is only set later when the middleware stack is resolved. 
I tried to apply a dirty hack that stores the config on the command so it can be accessed later on when we have the command name, but it felt very shaky and non reliable, therefore I added fallback to use the data we for sure expect to be available.

I agree that we prefer to use the `clientConfig.serviceId` if possible, but code should be prepared for it to be missing. In this case, we will produce non-consistent attributes from the instrumentation.

Unfortunately, I don't think there is anything we can do about it without any refactoring of the package by the aws-sdk v3 team, but I'm always open to fresh ideas and new directions.

Reply:
I see, thanks for telling me background on this 🙂 . So you mean that since you

> store[d] the config on the command so it can be accessed later on

it could be that the AWS SDK V3 changes it in a way that destroys it, so we should be safe and have a default in case it disappears in-between it being added and being accessed.

If I understood correctly, that makes sense to me 😄